zombies/src/main/java/com/github/phantazmnetwork/zombies/equipment/gun/shoot/fire/projectile/ProjectileFirer.java

```diff
@@ -33,14 +29,40 @@
 import java.lang.ref.Reference;
 import java.lang.ref.WeakReference;
 import java.util.*;
-import java.util.function.BiConsumer;
 import java.util.function.Supplier;
 
 /**
  * A {@link Firer} that works by firing projectiles.
  */
+@Model("zombies.gun.firer.projectile")
 public class ProjectileFirer implements Firer {
 
+    private static final ElementFactory<Data, ProjectileFirer> FACTORY = (objectData, context, dependencyProvider) -> {
+        EventNode<Event> node = dependencyProvider.provide(Key.key("zombies.dependency.event.instance_node"));
+
+        Supplier<Optional<? extends Entity>> entitySupplier =
+                dependencyProvider.provide(Key.key("zombies.dependency" + ".gun.shooter.supplier"));
+        UUID shooterUUID = dependencyProvider.provide(Key.key("zombies.dependency.gun.firer.projectile.shooter.uuid"));
+        ShotEndpointSelector endpointSelector =
+                context.provide(objectData.endSelectorPath(), dependencyProvider, false);
+        TargetFinder targetFinder = context.provide(objectData.targetFinderPath(), dependencyProvider, false);
+        ProjectileCollisionFilter collisionFilter =
+                context.provide(objectData.collisionFilterPath(), dependencyProvider, false);
+        Collection<ShotHandler> shotHandlers = new ArrayList<>(objectData.shotHandlerPaths().size());
+        for (String shotHandlerPath : objectData.shotHandlerPaths()) {
+            shotHandlers.add(context.provide(shotHandlerPath, dependencyProvider, false));
+        }
+        MobSpawner spawner = dependencyProvider.provide(Key.key("zombies.dependency.mob.spawner"));
+
+        ProjectileFirer firer =
+                new ProjectileFirer(objectData, entitySupplier, shooterUUID, endpointSelector, targetFinder,
+                        collisionFilter, shotHandlers, spawner);
+        node.addListener(ProjectileCollideWithBlockEvent.class, firer::onProjectileCollision);
+        node.addListener(ProjectileCollideWithEntityEvent.class, firer::onProjectileCollision);
```

Comment:
how are these listeners removed from the node?

Reply:
currently just expecting the event node to be dropped after the game is over since event nodes should be instance-based? or rather even game based